controllers/kustomization_generator.go

```diff
@@ -80,6 +80,8 @@ func (kg *KustomizeGenerator) WriteFile(dirPath string) (string, error) {
 		kus.Namespace = kg.kustomization.Spec.TargetNamespace
 	}
 
+	kus.Images = kg.kustomization.Spec.Images
```

Comment:
This removes images even if the override is for some of them. We need to merge the two arrays and override only the ones specified in `kustomization .Spec.Images`.